content/en/build/building-with-lotus-APIs.md

```diff
@@ -0,0 +1,646 @@
+
+## Query chain data
+
+Blockchain data is a fundamental part of any blockchain network. Using Lotus APIs, you can query chain data, including tipsets, blocks, messages, network status, storage deals, storage provider details, and more.
+
+Let's look at some common chain data-related packages in the filecoin.js library's LotusClient. 
+
+- `LotusClient.common` - methods to get node info
+- `LotusClient.chain` - methods to interact with the blockchain
+- `LotusClient.wallet` - methods to manage wallet on the Lotus node
+- `LotusClient.client` - methods to interact with the storage and retrieval markets as a client
+- `LotusClient.paych` - methods to interact with and manage payment channels
+- `LotusClient.state` - methods to query, inspect, and interact with chain state
+- more ...
+
+In the previous step, we already created a `chainDataQuery.mjs` file to demonstrate the basic steps to connect to the Lotus node and initialize a LotusClient to query chain data. Now let's write more code in there to learn other query functions that do not require auth-token to interact with the Lotus node.
+
+1. Get current chain head
+
+   This function will return the current head of the chain, including current block height, a list of blocks, and a list of CIDs.
+     ```javascript with-output
+      //Query the current block head
+      const chainHead = await lotusClient.chain.getHead();
+      console.log(chainHead);
+     ```
+
+      ```shell
+      {
+        Cids: [
+          {'/': 'bafy2bzacecdkonmhngylnnhrk4azkg2wkgcm6cnm5qn5sk4ww5cszjlvkgkd6'},
+          {'/': 'bafy2bzaceaglcpzhd5gfrzdyt7ce3e5asnbfz3s3stbqyxniziny5snewbpbg'}
+          ],
+        Blocks: [
+          {...},
+          {...}
+          ],
+        Height: 781268
+      }
+      ```
+
+2. Get messages in a block
+
+    ```javascript with-output
+    //Query all the messages in a block
+    const tipSet = await lotusClient.chain.getTipSetByHeight(781267);
+    const messages = await lotusClient.chain.getBlockMessages(tipSet.Cids[0]);
+    console.log(messages);
+    ```
+
+    ```shell
+    {
+      BlsMessages: [
+        {
+          Version: 0,
+          To: 't3ukhj6tpkgxjknu54opiaej2vrjz7nh7gzodkqlhpfphc6gxkogzmojv2cnlpgcuwkvnyhloctnc6lmlvceuq',
+          From: 't3ws4jhb7g4n5s7t3mwsc2enzfija2kzo3p7vkqdqbiaqrabqhmfxzzy3knrvn3ykp75okab6pse2nxud3ineq',
+          ...
+        }
+      ],
+      SecpkMessages: [ { Message: [Object], Signature: [Object], CID: [Object] } ],
+      Cids: [
+        {'/': 'bafy2bzaceapv6ms4m3x4gdaefmpilgsqfhfsrvrciw4iux4n7va47es4vudpa'},
+        {'/': 'bafy2bzaceaxaz5ews6a2jnwmeafhi2p5xpwjz7kogilakblp3q3wpxa3ax2xy'}
+      ]
+    }
+    ```
+
+3. Get wallet balance
+
+    With a given TestNet wallet address, you can query its balance by calling lotusClient.wallet.balance() function.
+
+    ```javascript
+    //Query Wallet balance
+    const walletBalance = await lotusClient.wallet.balance("t1ne72cbn6r55wea7ifjv4ypyti7t2df5dumsjhzq");
+    ```
```

Comment:
See previous comment about explaining your code _after_ you've shown it to the user.

Reply:
Fix it.

Reply:
Fix it.